### PR TITLE
test(policy-engine): restore FFI coverage

### LINKED ISSUE #3943: policy-engine: FFI test coverage dropped from 15 Rust tests to 1 after the retarget
core/tests/ffi_roundtrip.rs and ffi_default_dispatchers.rs (15 tests) were removed with the embedded engine; only assert_ffi_uses_opa in sdk/rust/tests/upstream_compatibility.rs remains. The .NET harness covers part of the surface. Restore roundtrip and default-dispatcher coverage against the pinned crate.

Found in the group review of the policy-engine retarget (#3939) and deferred from the follow-up (#3940).

### PR DESCRIPTION
**Pull request by Ricky-G**
## Summary

Restore the 15 Rust FFI regression tests removed during the policy-engine retarget, covering callback round trips, loader paths, malformed requests, transforms, identities, and default dispatchers.

## Problem

The retarget removed `core/tests/ffi_roundtrip.rs` and `core/tests/ffi_default_dispatchers.rs`, leaving only one FFI assertion in the compatibility suite.

## Changes

| File | What changed |
|------|-------------|
| `policy-engine/sdk/rust/tests/ffi_roundtrip.rs` | Restored 10 FFI callback, loader, transform, identity, and fail-closed request tests against the pinned SDK ABI. |
| `policy-engine/sdk/rust/tests/ffi_default_dispatchers.rs` | Restored 5 default-dispatcher and builder-limit tests, gated on `bundled-dispatchers` and `opa`. |

## Testing

- `cargo fmt --manifest-path policy-engine/Cargo.toml --all -- --check`
- `cargo clippy --locked --manifest-path policy-engine/Cargo.toml --workspace --all-targets -- -D warnings`
- `cargo test --locked --manifest-path policy-engine/Cargo.toml --workspace` with pinned OPA 0.70.0: passed
- `cargo test --locked --manifest-path policy-engine/Cargo.toml -p agent_control_specification --features bundled-dispatchers,opa`: passed, including all 15 restored FFI tests
- The optional Regorus CI-parity commands require Visual Studio Spectre-mitigated libraries unavailable on this Windows host; those commands are covered by the Ubuntu CI job.

Closes #3943

### CODE REVIEW

File: policy-engine/sdk/rust/tests/ffi_default_dispatchers.rs [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+        -1,
+        "null builder must fail closed"
+    );
+    let _ = take_err(err);
+
+    let yaml = CString::new(REGO_MANIFEST).unwrap();
+    let mut err: *mut c_char = ptr::null_mut();
+    let builder = unsafe { acs_builder_from_yaml(yaml.as_ptr(), &mut err) };
+    assert!(!builder.is_null(), "builder construction failed");
+    assert_eq!(
```
Comment (MohammadHaroonAbuomar) [inline policy-engine/sdk/rust/tests/ffi_default_dispatchers.rs:212]: `set_url_fetch_limits_validates_and_threads_through_build` asserts `acs_builder_set_url_fetch_limits(...) == 0`, but ffi.rs:531 returns -1 for every non-null builder since your #4025 merged yesterday. On the merged head with `--features bundled-dispatchers,opa` the test fails (`left: -1, right: 0`) and contradicts upstream_compatibility.rs:27, which pins -1 and the message. Drop the test (upstream_compatibility already covers it) or rewrite it to assert -1 plus the message, free the builder, and rename it to say "rejects".

File: policy-engine/sdk/rust/tests/ffi_default_dispatchers.rs [hunk 2]
```diff
@@ -0,0 +1,259 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#![cfg(all(feature = "bundled-dispatchers", feature = "opa"))]
```
Comment (MohammadHaroonAbuomar) [inline policy-engine/sdk/rust/tests/ffi_default_dispatchers.rs:4]: the file is gated on `#![cfg(all(feature = "bundled-dispatchers", feature = "opa"))]` and nothing in CI enables those features for this target: `cargo test --workspace` compiles it to a zero-test binary, and the one feature-enabled step (.github/ci/workflows.toml:75) runs `--test upstream_compatibility` only. So the five "restored" dispatcher tests never run in CI, which is why CI is green with a failing test inside. Add `[[test]] name = "ffi_default_dispatchers" required-features = ["bundled-dispatchers", "opa"]` to policy-engine/sdk/rust/Cargo.toml (same pattern as the `artifact_validation` entry at line 77), extend the workflows.toml step to run `--test ffi_default_dispatchers` too (or add a `cargo test --locked -p agent_control_specification --features bundled-dispatchers,opa` step), then `python3 scripts/ci/generate_workflows.py --write`.

**Review by MohammadHaroonAbuomar — CHANGES_REQUESTED** [review]
- PR body, Testing: "all 15 restored FFI tests" passed was true at the merge base 62c47122, before #4025, not at the head. Rerun `cargo test --locked -p agent_control_specification --features bundled-dispatchers,opa` on the merged head and update the body.
- Non-blocking: the merge commit 9b61af68 carries no Signed-off-by (DCO exempts merges; a rebase or the squash removes it). Also non-blocking: none of the ten roundtrip tests pass a null runtime, a null or non-UTF-8 request, `n = 0`, or a null `yamls` array; one extra test in ffi_roundtrip.rs covering those fail-closed branches would pin them (the same gap existed in the removed originals).

File: policy-engine/sdk/rust/tests/ffi_default_dispatchers.rs (hunk 2 above)
Comment (Ricky-G) [reply policy-engine/sdk/rust/tests/ffi_default_dispatchers.rs:4]: Registered `ffi_default_dispatchers` in `sdk/rust/Cargo.toml` with its required features, added it to the feature-enabled policy-engine CI step, and regenerated `policy-engine-ci.yml`. The CI command now executes all five dispatcher tests instead of compiling a zero-test binary.

<sub>Auto-replied by the [GitHub Copilot app](https://gh.io/app)</sub>

File: policy-engine/sdk/rust/tests/ffi_default_dispatchers.rs (hunk 1 above)
Comment (Ricky-G) [reply policy-engine/sdk/rust/tests/ffi_default_dispatchers.rs:215]: Updated the test to assert the pinned alpha.3 behavior: `acs_builder_set_url_fetch_limits` returns `-1` with the explicit unavailable-limits error, and the test now frees the builder. It is renamed to reflect the fail-closed rejection.

<sub>Auto-replied by the [GitHub Copilot app](https://gh.io/app)</sub>

**Review by MohammadHaroonAbuomar — APPROVED** [review]
Verified at 8da43b43: the dispatcher test now asserts the -1 that #4025 introduced and frees the builder, the file has a required-features entry so it is no longer a silent zero-test binary, and the OPA step runs it in CI (the log shows five tests passing). Ten roundtrip tests survived six mutation checks in the first round; 78 tests pass with default features and 83 with the features, fmt and clippy clean, workflows regenerated and up to date, body matches the runs. All checks green.

Non-blocking follow-ups: the CI clippy step runs without the features so the gated file is only clippy-checked locally, and ffi_roundtrip.rs still has no null-runtime, non-UTF-8 request, n=0 or null-yamls cases.
